internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go

```diff
@@ -854,6 +862,44 @@ func (c *ClusterManager) buildObservabilityCloudwatchLoggingCredentialsSecret()
 	}
 }
 
+func (c *ClusterManager) buildObservabilityEnterpriseCloudwatchLoggingCredentialsSecret(orgId string) *k8sCoreV1.Secret {
+	for _, enterpriseCredential := range c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.EnterpriseCredentials{
+		if orgId == enterpriseCredential.OrgId {
+			stringDataMap := map[string]string{
+				"aws_access_key_id": enterpriseCredential.AccessKey,
+				"aws_secret_access_key": enterpriseCredential.SecretAccessKey,
+			}
+
+			return &k8sCoreV1.Secret{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: k8sCoreV1.SchemeGroupVersion.String(),
+					Kind:       "Secret",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.K8sCredentialsSecretName,
+					Namespace: c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.K8sCredentialsSecretNamespace,
+				},
+				Type:       k8sCoreV1.SecretTypeOpaque,
+				StringData: stringDataMap,
+			}
+		}
+	}
```

Comment:
I think this would be more readable using the `arrays utilities:

```suggestion
idx, enterpriseCredential := arrays.FindFirst(c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.EnterpriseCredentials, func(enterpriseCredential observatorium.ObservabilityEnterpriseCloudwatchLoggingConfigCredentials) bool {
		return orgId == enterpriseCredential.OrgId
	})

	if idx != arrays.ElementNotFound {
		stringDataMap := map[string]string{
			"aws_access_key_id":     enterpriseCredential.AccessKey,
			"aws_secret_access_key": enterpriseCredential.SecretAccessKey,
		}

		return &k8sCoreV1.Secret{
			TypeMeta: metav1.TypeMeta{
				APIVersion: k8sCoreV1.SchemeGroupVersion.String(),
				Kind:       "Secret",
			},
			ObjectMeta: metav1.ObjectMeta{
				Name:      c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.K8sCredentialsSecretName,
				Namespace: c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.K8sCredentialsSecretNamespace,
			},
			Type:       k8sCoreV1.SecretTypeOpaque,
			StringData: stringDataMap,
		}
	}
```

Reply:
:+1: